### ISSUE TITLE
issue on Apple M2 chip MacBook, which uses arm64

### ISSUE THREAD

**Issue by MuhammadHassanShah** [mapbox/mapbox-search-ios#234]
Failed to build module 'MapboxSearch' for importation due to the errors above; the textual interface may be broken by project issues or a compiler bug

<img width="1389" alt="Screenshot 2024-05-09 at 11 21 04 AM" src="https://github.com/mapbox/mapbox-search-ios/assets/106577849/3d533d03-7d91-4823-aefa-9a37906673a0">

**Comment by MuhammadHassanShah** [comment]
@zugaldia  @mourner  @sbma44  @Gionfriddo    please suggest us solution

**Comment by aokj4ck** [comment]
@MuhammadHassanShah what version of the SDK are you currently using?

If you are using a v2 release then please update to our latest release v2.0.3 and let me know if you encounter any issues. https://github.com/mapbox/mapbox-search-ios/releases/tag/v2.0.3

**Comment by rgmyyw** [comment]
this problem persists

```ruby
  pod 'MapboxSearch', "2.0.3"
  pod 'MapboxSearchUI', '2.0.3'
  pod 'MapboxCommon', '24.3.1'
  pod 'MapboxMaps', '11.3.0'
  pod 'MapboxStatic.swift', '0.12.0'
  pod 'MapboxDirections' ,'2.7.0'
```

Xcode:  Xcode15.4 `15F31d`

**Comment by aokj4ck** [comment]
@MuhammadHassanShah @rgmyyw I am working on resolving this for you.

**Comment by aokj4ck** [comment]
@MuhammadHassanShah @rgmyyw Are you able to update to MapboxCommon 24.4.0, MapboxSearch/SearchUI 2.1.0, and MapboxMaps 11.4.0?

This can be fixed in MapboxSearch version 2.1.0 by adding `pod 'MapboxCoreSearch', '2.1.0'`. I am working on correcting the dependencies in the next release so that this extra pod is imported automatically.

```
  pod "MapboxSearch", '2.1.0'
  pod "MapboxSearchUI", '2.1.0'
  pod "MapboxCoreSearch", '2.1.0'  # NEW

  pod 'MapboxCommon', '24.4.0'
  pod 'MapboxMaps', '11.4.0'
  pod 'MapboxStatic.swift', '0.12.0'
  pod 'MapboxDirections' ,'2.7.0'
```

**Comment by aokj4ck** [comment]
@MuhammadHassanShah @rgmyyw Previously I shared a work-around by importing an additional pod. That workaround is no longer necessary when you upgrade to MapboxSearch v2.1.1 / MapboxCommon 24.4 / MapboxMaps 11.4. I'm sorry for the disruption and please let me know if this resolves the issue for you in this thread, and open new issues for any other problem you encounter.

Release v2.1.1: https://github.com/mapbox/mapbox-search-ios/releases/tag/v2.1.1
